plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/reload/SpringChangedReloadCommand.java

```diff
@@ -0,0 +1,54 @@
+package org.hotswap.agent.plugin.spring.reload;
+
+import org.hotswap.agent.command.MergeableCommand;
+import org.hotswap.agent.logging.AgentLogger;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class SpringChangedReloadCommand extends MergeableCommand {
+    private static AgentLogger LOGGER = AgentLogger.getLogger(XmlsChangedCommand.class);
+
+    // unit test only
+    private static AtomicLong waitingTaskCount = new AtomicLong(0);
+
+    ClassLoader appClassLoader;
+    long timestamps;
+
+    public SpringChangedReloadCommand(ClassLoader appClassLoader) {
+        this.appClassLoader = appClassLoader;
+        this.timestamps = System.currentTimeMillis();
+        LOGGER.trace("SpringChangedReloadCommand created with timestamp '{}'", timestamps);
+        waitingTaskCount.incrementAndGet();
+    }
+
+    @Override
+    public void executeCommand() {
+        // async call to avoid reload too much times
+        // FIXME use threadpool ?
+        new Thread(() -> {
```

Comment:
This looks not good to me. I am not sure why it can avoid of reloading multiple times by firing a new thread. IMO, we should make this command a real mergable command so that no matter how many commands are scheduled, there's only one command running on behalf of others.

What we need to do is to implement an equals method

```java
    @Override
    public boolean equals(Object object) {
        if (this == object) return true;
        if (object == null || getClass() != object.getClass()) return false;
        SpringChangedReloadCommand that = (SpringChangedReloadCommand) object;
        return Objects.equals(appClassLoader, that.appClassLoader);
    }

    @Override
    public int hashCode() {
        return Objects.hash(appClassLoader);
    }
```

Reply:
Remove the thread.